test/hotspot/jtreg/compiler/lib/ir_framework/driver/irmatching/parser/VMInfo.java

```diff
@@ -0,0 +1,119 @@
+        TestFramework.check(isKey("MaxVectorSizeIsDefault"), "VMInfo does not contain MaxVectorSizeIsDefault");
+        TestFramework.check(isKey("LoopMaxUnroll"), "VMInfo does not contain LoopMaxUnroll");
+        TestFramework.check(isKey("UseAVX"), "VMInfo does not contain UseAVX");
+        TestFramework.check(isKey("UseAVXIsDefault"), "VMInfo does not contain UseAVXIsDefault");
+        System.err.println("--- VMInfo from Test VM ---");
+        System.err.println("cpuFeatures:   " + getStringValue("cpuFeatures"));
+        System.err.println("MaxVectorSize: " + getLongValue("MaxVectorSize"));
+        System.err.println("MaxVectorSizeIsDefault: " + getLongValue("MaxVectorSizeIsDefault"));
+        System.err.println("LoopMaxUnroll: " + getLongValue("LoopMaxUnroll"));
+        System.err.println("UseAVX: " + getLongValue("UseAVX"));
+        System.err.println("UseAVXIsDefault: " + getLongValue("UseAVXIsDefault"));
+        if (isDefaultCascadeLake()) {
+            System.err.println(" -> You are on default Cascade Lake");
+            System.err.println("   -> SuperWord expected to run with 32 byte, not 64 byte, VectorAPI expected to use 64 byte");
+            System.err.println("   -> \"canTrustVectorSize == false\", some vector node IR rules are made weaker.");
+        }
+    }
+
+    public String getStringValue(String key) {
+        TestFramework.check(isKey(key), "VMInfo does not contain \"" + key + "\"");
+        return keyValueMap.get(key);
+    }
+
+    public long getLongValue(String key) {
+        try {
+            return Long.parseLong(getStringValue(key));
+        } catch (NumberFormatException e) {
+            throw new TestFrameworkException("VMInfo value for \"" + key + "\" is not a long, got \"" + getStringValue(key) + "\"");
+        }
+    }
+
+    public boolean hasCPUFeature(String feature) {
+        String features = getStringValue("cpuFeatures") + ",";
+        return features.contains(" " + feature + ",");
+    }
+
+    public boolean isCascadeLake() {
+        Matcher matcher = CPU_SKYLAKE_PATTERN.matcher(getStringValue("cpuFeatures"));
+        if (!matcher.find()) {
+            return false; // skylake pattern not found
+        }
+        String stepping = matcher.group(1).trim();
+        return Long.parseLong(stepping) >= 5; // this makes it Cascade Lake
+    }
+
+    public boolean isDefaultCascadeLake() {
+        // See VM_Version::is_default_intel_cascade_lake
+        return isCascadeLake() &&
+               getLongValue("MaxVectorSizeIsDefault") == 1 &&
+               getLongValue("UseAVXIsDefault") == 1 &&
+               getLongValue("UseAVX") > 2;
+    }
+
+    /**
+     * Some platforms do not behave as expected, and one cannot trust that the vectors
+     * make use of the full MaxVectorSize. For Cascade Lake we by default only use
+     * 32 bytes for SuperWord even though MaxVectorSize is 64. But the VectorAPI still
```

Comment:
Suggestion: For Cascade Lake, we only use 32 bytes for SuperWord by default even though MaxVectorSize is 64.